cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java

```diff
@@ -67,83 +118,135 @@ private Organization fromSubSubunit(SubSubUnitDto subSubUnitDto) {
+        if (newEffort.isFailure()) {
+            logger.warn(String.format("Failed HttpRequest on attempt %d of 3: ", effortCount + 1)
+                    + newEffort.getException().getMessage(), newEffort.getException()
+            );
+        }
+        return newEffort;
+    }
+
+    private CompletableFuture<HttpResponse<String>> createAndSendHttpRequest(URI uri) {
+        HttpRequest httpRequest = HttpRequest.newBuilder()
                 .GET()
                 .uri(uri)
                 .build();
+        return httpClient.sendAsync(httpRequest, BodyHandlers.ofString());
     }
 
-    protected HttpResponse<String> sendRequest(HttpRequest httpRequest) {
-        return attempt(() -> httpClient.sendAsync(httpRequest, BodyHandlers.ofString()).get()).orElseThrow();
+    private boolean shouldTryMoreTimes(int effortCount) {
+        return effortCount < MAX_EFFORTS;
     }
 
-    private boolean isSuccessful(int statusCode) {
-        return statusCode <= HTTP_MULT_CHOICE && statusCode >= HTTP_OK;
+    @SuppressWarnings("PMD.UselessParentheses") // keep the parenthesis for clarity
+    private boolean shouldKeepTrying(int effortCount, Try<HttpResponse<String>> lastEffort) {
+        return lastEffort == null || (lastEffort.isFailure() && shouldTryMoreTimes(effortCount));
+    }
+
+    private int waitBeforeRetrying(int effortCount) {
+        if (effortCount > FIRST_EFFORT) {
+            try {
+                Thread.sleep(WAITING_TIME);
+            } catch (InterruptedException e) {
+                logger.error(LOG_INTERRUPTION);
+                throw new RuntimeException(e);
+            }
+        }
+        return effortCount;
+    }
+
+    protected SubSubUnitDto getSubSubUnitDtoWithMultipleEfforts(URI subunitUri)
+            throws ApiGatewayException {
+
+        SubSubUnitDto subsubUnitDto = Try.of(subunitUri)
+                .flatMap(this::sendRequestMultipleTimes)
+                .map((HttpResponse<String> response) -> throwExceptionIfNotSuccessful(response, subunitUri))
+                .map(HttpResponse::body)
+                .map(SubSubUnitDto::fromJson)
+                .orElseThrow(this::handleError);
+
+        subsubUnitDto.setSourceUri(subunitUri);
+        return subsubUnitDto;
     }
 
+    protected HttpResponse<String> throwExceptionIfNotSuccessful(HttpResponse<String> response, URI requestedUri)
+            throws FailedHttpRequestException, NotFoundException {
+        if (isNull(response)) {
+            throw new FailedHttpRequestException(NULL_HTTP_RESPONSE_ERROR_MESSAGE);
+        } else if (response.statusCode() >= FIRST_SUCCESSFUL_CODE
+                && response.statusCode() < FIRST_NON_SUCCESSFUL_CODE) {
+            return response;
+        } else if (response.statusCode() == HTTP_NOT_FOUND) {
+            throw new NotFoundException(String.format(NOT_FOUND_MESSAGE_TEMPLATE, requestedUri));
+        } else {
+            throw new FailedHttpRequestException(errorMessage(response));
```

Comment:
Error messages is used internally right? I can see you throw a client ApiGatewayException in handleError so I assume this is internal